include/9on12Util.h

```diff
@@ -35,6 +35,61 @@ namespace D3D9on12
         }
     };
 
+    class Result
+    {
+    public:
+        enum Value : uint8_t
+        {
+            S_SUCCESS = 0x00,
+            S_CHANGE = 0x01,
+            E_FAILURE = 0x10,
+            E_INVALID_ARG = 0x20
+        };
+
+        Result() = default;
+        constexpr Result(Value result) : value(result) {}
+
+        //Prevent usage: if(result)
+        explicit operator bool() = delete;

```

Comment:
Do you need to `delete` this? I'd expect just not defining it would be sufficient.

Reply:
Ah thanks, that was left over from before I switched to using AsEnum() instead of allowing Result to be converted to a Value enum for switches. Removing this and the equally useless HRESULT one.